File: iOS7bookExamples/bk2ch06p313pageControllerBug/ch19p628pageControllerBug/MyPage.m

#import "MyPage.h"

@interface MyPage ()
@property (weak, nonatomic) IBOutlet UILabel *lab;
@end

@implementation MyPage

- (void) viewDidLoad {
    [super viewDidLoad];
    self.lab.text = [NSString stringWithFormat:@"%ld",(long)self.num];
}

- (void) setNum: (NSInteger) newNum {
    _num = newNum;
    self.lab.text = [NSString stringWithFormat:@"%ld",(long)newNum];
}

@end

File: bk2ch06p313pageControllerBug/ch19p628pageControllerBug/MyPage.swift


import UIKit

class MyPage : UIViewController {
    @IBOutlet var lab : UILabel!
    var num : Int = 0 {
    didSet {
        if let lab = self.lab {
            lab.text = "\(self.num)"
        }
    }
    }
    
    init() {
        super.init(nibName: "MyPage", bundle: nil)
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }

    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.lab.text = "\(self.num)"
    }
}
